File: platforms/ios/foundation/Sources/NativeblocksFoundation/block/Box.swift
import NativeblocksCompiler
import NativeblocksRuntime
import SwiftUI

@Block(
    name: "Box",
    keyType: "nativeblocks/box",
    description: "Stacking layout container; style it by attaching modifiers.",
    version: 1,
    versionName: "1"
)
struct Box<Content: View>: View {
    var blockContext: BlockContext? = nil

    @BlockData(
        description: "Children alignment (topStart, topCenter, topEnd, centerStart, center, centerEnd, bottomStart, bottomCenter, bottomEnd).",
        defaultValue: "topStart"
    )
    var contentAlignment: Alignment = .topLeading

    @BlockSlot(description: "Slot for composing child content within the box.")
    var content: (BlockIndex) -> Content

    var body: some View {
        ZStack(alignment: contentAlignment) {
            content(-1)
        }
        .modifier(blockContext?.modifier ?? .none)
    }
}

File: platforms/android/foundation/foundation/src/main/java/io/nativeblocks/foundation/block/Box.kt
package io.nativeblocks.foundation.block

import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import io.nativeblocks.compiler.type.Block
import io.nativeblocks.compiler.type.BlockData
import io.nativeblocks.compiler.type.BlockIndex
import io.nativeblocks.compiler.type.BlockSlot
import io.nativeblocks.runtime.api.provider.block.BlockContext
import io.nativeblocks.runtime.api.provider.block.NONE_INDEX
import androidx.compose.foundation.layout.Box as ComposeBox

@Block(
    keyType = "nativeblocks/box",
    name = "Box",
    description = "Stacking layout container; style it by attaching modifiers.",
    version = 1,
    versionName = "1",
)
@Composable
internal fun Box(
    blockContext: BlockContext,
    @BlockData(
        description = "Children alignment (topStart, topCenter, topEnd, centerStart, center, " +
            "centerEnd, bottomStart, bottomCenter, bottomEnd).",
        defaultValue = "topStart",
    ) contentAlignment: Alignment = Alignment.TopStart,
    @BlockSlot(
        description = "Slot for composing child content within the box.",
    ) content: @Composable (index: BlockIndex, scope: Any) -> Unit,
) {
    ComposeBox(
        modifier = blockContext.modifier,
        contentAlignment = contentAlignment,
    ) {
        content(NONE_INDEX, this)
    }
}
